### ISSUE TITLE
Restructure src folder - Align namespace & directories

### ISSUE THREAD

**Issue by vmuriart** [pythonnet/pythonnet#440]
### Environment

-   Pythonnet version: All
-   Python version: All
-   Operating System: All

tl;dr: Rename/Relocate folders. Fix namespaces.

### Details

Our `src` and `tests` folders are disorganized and a bit counter-intuitive. 

- folder: `src\tests` 
  namespace: `n\a`
  purpose: contains python tests
  Issue: `pytest` has issue with detecting certain configuration if the tests aren't located on the root folder. https://github.com/pytest-dev/pytest/issues/2257
- folder: `src\testing` 
  namespace: `Python.Test` 
  purpose: Fixture for Python tests. 
  Issue: Name and namespace seem to indicate that embedded_tests would be located here. Should rename folder and namespace to something more appropriate. `Python.Fixtures` maybe.
- folder: `src\embed_tests`, 
  namespace: `Python.EmbeddingTest`, 
  purpose: C# tests. 
  Issue: Namespace and folder don't match. confusing with `Python.Test`. May need to be relocated if we relocate `src\tests` for consistency purposes
- folder: `src\console`
  namespace: `Python.Runtime`
  purpose: Example of embedding python in c#. Troubleshooting.
  Issue: Wrong namespace since it isn't part of `Python.Runtime` at all. change to `Python.Example` maybe?
- folder: `src\clrmodule`
  namespace: `None`
  purpose: `clr.pyd`
  Issue: No namespace. 
- folder: `src\monoclr`
  namespace: `N/A`
  purpose: mono clr
  Issue: folder name could make it more obvious that its related to `src\clrmodule`. Could also add a project file to make it visible in the `sln`
- folder: `src\runtime`
  namespace: `Python.Runtime`
  purpose: All mighty project
  Issue: Class `Runtime` clashes with namespace making it a pain to reference when testing against it. For now I don't think there's much we can do since the class is `Public` and renaming the class or namespace is a major breakage. Just wanted to mention it in case someone can think of a way of renaming it without breaking backwards compatibility. 

Also in general, I would like to propercase the filenames to match the classes they contain. I started doing this with the new classes/tests but avoided touching existing ones. for example `pyint.cs` would become `PyInt.cs`.

I can take care of all the changes, but wanted to agree on a design/pattern before doing any changes.

**Comment by vmuriart** [comment]
ok with delaying changes with `src\runtime` until `3.0.0`. All others we can safety rename/relocate without issues.

**Comment by denfromufa** [comment]
- I agree that all these tests directories are super-confusing, I still don't remember what type of tests are located in these 3 folders.

- clrmodule may have been designed without namespace for dllexport to work?

- console can be placed to nPython or Python.Console namespace.

- agree on monoclr.

- Python.Runtime should probably be backwards compatible as much as possible since it is public.

- i don't have strong opinion on pyint.cs

**Comment by vmuriart** [comment]
> clrmodule may have been designed without namespace for dllexport to work?

I added a namespace to it and its worked with no issues. Since the `clrmodule` was added later on, it was probably an oversight. 

> console can be placed to nPython or Python.Console namespace.

I like the name `Python.Example.Console`

> Python.Runtime should probably be backwards compatible as much as possible since it is public.

Should it be public though? Most methods are `private`/`internal`. 

> i don't have strong opinion on pyint.cs

I meant `PyInt.cs` as an example. It would apply to all files.

**Comment by filmor** [comment]
This is done.
